src/spline2/MultiBsplineEval_helper.hpp

```diff
@@ -20,74 +20,37 @@
 #include <algorithm>
 #include "bspline_traits.hpp"
 #include "spline2/MultiBsplineData.hpp"
+#include "Numerics/SplineBound.hpp"
 
 namespace spline2
 {
 
-/** break x into the integer part and residual part and apply bounds
- * @param x input coordinate
- * @param dx fractional part
- * @param ind integer part
- * @param nmax upper bound of the integer part
- *
- * x in the range of [0, nmax+1) will be split correctly.
- * x < 0, ind = 0, dx = 0
- * x >= nmax+1, ind = nmax, dx = 1 - epsilon
- *
- * Attention: nmax is not the number grid points but the maximum allowed grid index
- * For example, ng is the number of grid point.
- * the actual grid points indices are 0, 1, ..., ng - 1.
- * In a periodic/anti periodic spline, set nmax = ng - 1
- * In a natural boundary spline, set nmax = ng - 2
- * because the end point should be excluded and the last grid point has an index ng - 2.
- */
-template<typename T, typename TRESIDUAL>
-inline void getSplineBound(T x, TRESIDUAL& dx, int& ind, int nmax)
-{
-  // lower bound
-  if (x < 0)
-  {
-    ind = 0;
-    dx  = T(0);
-  }
-  else
-  {
-#if defined(__INTEL_LLVM_COMPILER) || defined(__INTEL_CLANG_COMPILER)
-    T ipart = std::floor(x);
-    dx = x - ipart;
-#else
-    T ipart;
-    dx  = std::modf(x, &ipart);
-#endif
-    ind = static_cast<int>(ipart);
-    // upper bound
-    if (ind > nmax)
-    {
-      ind = nmax;
-      dx  = T(1) - std::numeric_limits<T>::epsilon();
-    }
-  }
-}
-
 /** define computeLocationAndFractional: common to any implementation
  * compute the location of the spline grid point and residual coordinates
  * also it precomputes auxiliary array a, b and c
  */
 template<typename T>
-inline void computeLocationAndFractional(const typename qmcplusplus::bspline_traits<T, 3>::SplineType* restrict spline_m,
-                            T x, T y, T z,
-                            int& ix, int& iy, int& iz,
-                            T a[4], T b[4], T c[4])
+inline void computeLocationAndFractional(
```

Comment:
is this what clang-format produced?  If this is hand formatting please protect it with format on and off tags.

Reply:
Done by clang-format